api/src/main/java/org/openmrs/api/db/hibernate/HibernateOrderDAO.java

```diff
@@ -274,6 +274,36 @@ public OrderGroup getOrderGroupByUuid(String uuid) throws DAOException {
 	public OrderGroup getOrderGroupById(Integer orderGroupId) throws DAOException {
 		return (OrderGroup) sessionFactory.getCurrentSession().get(OrderGroup.class, orderGroupId);
 	}
+
+	/**
+	 * @see OrderDAO#getOrderGroupsByPatient(Patient)
+	 */
+	@Override
+	public List<OrderGroup> getOrderGroupsByPatient(Patient patient) {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(OrderGroup.class);
+		if (patient != null) {
+			criteria.add(Restrictions.eq("patient", patient));
+		}
+		else{
+			criteria.add(Restrictions.isNull("patient"));
```

Comment:
Can we have an order group without a patient?

Reply:
I believe that patient and encounter are required for order group. What would you recommend @dkayiwa if someone calls this method with a null parameter. Should I throw an exception or assume that the precondition is that this method will not be called by a null value.